### PR TITLE
Telemetry update & Initialization Validations

### PR DESCRIPTION
**Pull request by elopezanaya**
# PR Details

### Description
_Include a description of the problem to be solved_

- Some public API's doenst have implemented the Failed Event
- Missing validations to prevent exposed API;s to be called before SDK initialization

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

- Adding missing functionality and telemetry

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -1560,7 +1610,8 @@ class OmnichannelChatSDK {
 
                 this.scenarioMarker.failScenario(TelemetryEvent.UploadFileAttachment, {
                     RequestId: this.requestId,
-                    ChatId: this.chatToken.chatId as string
+                    ChatId: this.chatToken.chatId as string,
+                    ExceptionDetails: JSON.stringify(error)
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:1614]: Are we sure we won't be logging message contents here by accident?

File: src/OmnichannelChatSDK.ts [hunk 2]
```diff
@@ -1709,7 +1769,8 @@ class OmnichannelChatSDK {
             console.error(`OmnichannelChatSDK/emailLiveChatTranscript/error: ${error}`);
             this.scenarioMarker.failScenario(TelemetryEvent.EmailLiveChatTranscript, {
                 RequestId: requestId,
-                ChatId: chatId
+                ChatId: chatId,
+                ExceptionDetails: JSON.stringify(error)
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:1773]: Are we sure we won't be logging message contents here by accident?

File: src/OmnichannelChatSDK.ts [hunk 3]
```diff
@@ -1190,7 +1219,8 @@ class OmnichannelChatSDK {
             } catch (error) {
                 this.scenarioMarker.failScenario(TelemetryEvent.SendMessages, {
                     RequestId: this.requestId,
-                    ChatId: this.chatToken.chatId as string
+                    ChatId: this.chatToken.chatId as string,
+                    ExceptionDetails: JSON.stringify(error)
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:1223]: Are we sure we won't be logging message contents here by accident?

**Review by mithunjohny — APPROVED** [review]

**Review by matan-hershberg-nuance — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Update of missing telemetry events for exposed API's
+- Adding exception details for telemetry for upload attachments
```
Comment (mithunjohny) [inline CHANGELOG.md:10]: we can update the log since you removed the exception details in telemetry

**Review by matan-hershberg-nuance — APPROVED** [review]

**Review by mithunjohny — APPROVED** [review]
